File: ruby/command-t/lib/command-t/metadata/fallback.rb
# SPDX-FileCopyrightText: Copyright 2015-present Greg Hurrell and contributors.
# SPDX-License-Identifier: BSD-2-Clause

module CommandT
  module Metadata
    # This file gets overwritten with actual data during the installation
    # process (when `ruby extconf.rb` is run).
    EXPECTED_RUBY_VERSION = '[unknown]'
    EXPECTED_RUBY_PATCHLEVEL = '[unknown]'
    UNKNOWN = true
  end
end

File: lua/wincent/commandt/private/finders/fallback.lua
-- SPDX-FileCopyrightText: Copyright 2022-present Greg Hurrell and contributors.
-- SPDX-License-Identifier: BSD-2-Clause

-- A lazy fallback wrapper (to avoid the cost of scanning until actually needed)
-- around the built-in file finder.
return function(finder, directory, options)
  return function()
    finder.fallback = require('wincent.commandt.private.finders.file')(directory ~= '' and directory or '.', options)
    return finder.fallback, 'fallback'
  end
end
